include/kernel.h

```diff
@@ -538,7 +538,7 @@ struct k_thread {
 
 #if defined(CONFIG_THREAD_NAME)
 	/* Thread name */
-	const char *name;
+	char name[CONFIG_THREAD_MAX_NAME_LEN];
```

Comment:
So, now, it looks to me that we require 32 bytes for each of the thread, while, before, this was on a per-thread basis, is this correct?

Reply:
correct

Reply:
because of that some tests are failing now on master: e.g. https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/45717/3/tests
That is because of increased RAM usage.

I wonder if array approach shouldn't be applied only in case of user space? In most of the cases users sets name like `k_thread_name_set(t, "thread_name")`. In that case ram buffer is a waste.

Reply:
hrm, the code used for thread awareness stores the offset to [`name`](https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/debug/openocd.c#L71). 

I don't see an easy way to resolve from the gdb server whether name is a pointer or an array (which changes how the memory needs to be read at the offset). If the array is kept, might be useful to encode that info in some way in the `.openocd_dbg` section